### PR TITLE
Disable semantic segmentation soft prediction processing (by default)

### PR DESCRIPTION
**Pull request by negvet**
### Summary
Fixing https://jira.devtools.intel.com/browse/CVS-114142

Similar to https://github.com/openvinotoolkit/training_extensions/pull/2217

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by sovrasov — DISMISSED** [review]

File: otx/algorithms/segmentation/adapters/openvino/task.py [hunk 1]
```diff
@@ -261,14 +263,15 @@ def add_prediction(
                     if label_index == 0:
                         continue
                     current_label_soft_prediction = soft_prediction[:, :, label_index]
-                    class_act_map = get_activation_map(current_label_soft_prediction)
+                    if process_soft_prediction:
+                        current_label_soft_prediction = get_activation_map(current_label_soft_prediction)
```
Comment (sovrasov) [inline otx/algorithms/segmentation/adapters/openvino/task.py:267]: Should this be done for pytorch task as well?

Comment (negvet) [reply otx/algorithms/segmentation/adapters/openvino/task.py:267]: Currently, torch sem segm task doesn't dump soft predictions

Comment (sovrasov) [reply otx/algorithms/segmentation/adapters/openvino/task.py:267]: https://github.com/openvinotoolkit/training_extensions/blob/3717b7245c8edbdcae45b57a4fb136d75031adef/otx/algorithms/segmentation/task.py#L291

Comment (jaegukhyun) [reply otx/algorithms/segmentation/adapters/openvino/task.py:267]: @negvet Could you check above mention?

**Review by sungmanc — CHANGES_REQUESTED** [review]
Please hold this a moment, Need to consider the release

File: otx/algorithms/segmentation/adapters/openvino/task.py (hunk 1 above)
Comment (negvet) [reply otx/algorithms/segmentation/adapters/openvino/task.py:267]: Indeed, will fix, thanks!

**Comment by negvet** [comment]
Open a new one

**Pull request closed without merging**